langchain/schema/agents.py

```diff
@@ -0,0 +1,18 @@
+from __future__ import annotations
+
+from typing import NamedTuple, Union
+
+
+class AgentAction(NamedTuple):
```

Comment:
nit: NamedTuples are python v2 in my mind, python 3 dataclasses with frozen=True is nicer for usage